pkg/csi/cinder/nodeserver.go

```diff
@@ -411,6 +411,24 @@ func (ns *nodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		}
 	}
 
+	// Try expanding the volume if it's created from a snapshot or another volume
+	if vol.SourceVolID != "" || vol.SnapshotID != "" {
+
+		r := mountutil.NewResizeFs(ns.Mount.Mounter().Exec)
+
+		needResize, err := r.NeedResize(devicePath, stagingTarget)
+
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not determine if volume %q need to be resized: %v", volumeID, err)
+		}
+
+		if needResize {
```

Comment:
can we add a line log to indicate we are going to resize here?

Reply:
sure, added 👍